zeus/monitor/energy.py

```diff
@@ -199,22 +199,27 @@ def _get_instant_power(self) -> tuple[dict[int, float], float]:
         power_measurement_time = time() - power_measurement_start_time
         return power, power_measurement_time
 
-    def begin_window(self, key: str, sync_cuda: bool = True) -> None:
+    def begin_window(
+        self, key: str, sync_cuda: callable[[], None] | bool = False
+    ) -> None:
         """Begin a new measurement window.
 
         Args:
             key: Unique name of the measurement window.
-            sync_cuda: Whether to synchronize CUDA before starting the measurement window.
-                (Default: `True`)
+            sync_cuda: if False, no CUDA syncrhonization. If it's a `True`, try to infer
```

Comment:
This needs an actual description assuming people have no idea what this argument does.